components/dashboard/src/projects/ProjectVariables.tsx

```diff
@@ -0,0 +1,132 @@
+                            <ItemField className="flex justify-end">
+                                <ItemFieldContextMenu menuEntries={[
+                                    {
+                                        title: 'Delete',
+                                        customFontStyle: 'text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300',
+                                        onClick: () => deleteEnvVar(variable.id),
+                                    },
+                                ]} />
+                            </ItemField>
+                        </Item>
+                    })}
+                </ItemsList>
+            </>
+        }
+    </ProjectSettingsPage>;
+}
+
+function AddVariableModal(props: { project?: Project, onClose: () => void }) {
+    const [ name, setName ] = useState<string>("");
+    const [ value, setValue ] = useState<string>("");
+    const [ censored, setCensored ] = useState<boolean>(true);
+    const [ error, setError ] = useState<Error | undefined>();
+
+    const addVariable = async () => {
+        if (!props.project) {
+            return;
+        }
+        try {
+            await getGitpodService().server.setProjectEnvironmentVariable(props.project.id, name, value, censored);
+            props.onClose();
+        } catch (err) {
+            setError(err);
+        }
+    }
+
+    return <Modal visible={true} onClose={props.onClose} onEnter={() => { addVariable(); return false; }}>
+        <h3 className="mb-4">Add Variable</h3>
+        <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 flex flex-col">
+            <AlertBox><strong>Project variables might be accessible by anyone with read access to your repository.</strong><br/>Secret values can be exposed if they are printed in the logs, persisted to the file system, or made visible in workspaces.</AlertBox>
+            {error && <div className="bg-gitpod-kumquat-light rounded-md p-3 text-gitpod-red text-sm mb-2">
+                {error}
+            </div>}
+            <div className="mt-8">
+                <h4>Name</h4>
+                <input autoFocus className="w-full" type="text" name="name" value={name} onChange={e => setName(e.target.value)} />
+            </div>
+            <div className="mt-4">
+                <h4>Value</h4>
+                <input className="w-full" type="text" name="value" value={value} onChange={e => setValue(e.target.value)} />
+            </div>
+            <div className="mt-4">
+                <CheckBox title="Hide in Workspaces" desc="Project variables are visible during prebuilds. Choose whether this variable should be visible in workspaces as well." checked={censored} onChange={() => setCensored(!censored)} />
+            </div>
+            {!censored && <div className="mt-4">
+                <InfoBox>This value will be directly visible to anyone who can open your repository in Gitpod.</InfoBox>
```

Comment:
**thought**: The layout change that happens on click could be quite abrupt and surpising for users. I'd suggest to not include this alert all together and default to having variables visible everywhere by default as described in https://github.com/gitpod-io/gitpod/pull/7295#discussion_r785052457. 💭

Reply:
Let's leave it in - i.o.w. not make it easier for users to leak secrets. 

There is definitely a scrolling problem on landscape low-height displays. (the bottom of form becomes unreachable)
![Screenshot 2022-01-14 at 20 08 56](https://user-images.githubusercontent.com/849592/149578715-f97512a7-8609-4778-b7b6-efe0d286797b.png)